openassessment/xblock/staff_assessment_mixin.py

```diff
@@ -0,0 +1,112 @@
+"""
+A mixin for staff grading.
+"""
+import logging
+
+from staff_area_mixin import require_course_staff
+from xblock.core import XBlock
+
+from openassessment.assessment.api import staff as staff_api
+from openassessment.workflow import api as workflow_api
+from openassessment.assessment.errors import (
+    StaffAssessmentRequestError, StaffAssessmentInternalError
+)
+
+from .data_conversion import create_rubric_dict
+from .resolve_dates import DISTANT_FUTURE
+from .data_conversion import clean_criterion_feedback, create_submission_dict
+
+logger = logging.getLogger(__name__)
+
+
+class StaffAssessmentMixin(object):
+    """
+    This mixin is for all staff-assessment related endpoints.
+    """
+
+    @XBlock.json_handler
+    @require_course_staff("STUDENT_INFO")
+    def staff_assess(self, data, suffix=''):
+        """
+        Create a staff assessment from a staff submission.
+        """
+        if 'options_selected' not in data:
```

Comment:
Can we share some of this validation logic with the other endpoints? They all seem to be doing the same thing (except that some of them don't require a submission_uuid.